prow/cmd/job-forker/forking.go

```diff
@@ -0,0 +1,278 @@
+		periodics = append(periodics, generatePeriodics(j, version, repo)...)
+	}
+	targetDir := path.Join(baseRepoJobsDirPath, forkDir)
+	newFileName := repoString + "-" + fileVersion + ".yaml"
+
+	output := path.Join(targetDir, filepath.Base(newFileName))
+
+	payload := make(map[string]interface{})
+
+	if len(presubmits) != 0 {
+		payload["presubmits"] = presubmits
+	}
+
+	if len(postsubmits) != 0 {
+		payload["postsubmits"] = postsubmits
+	}
+
+	if len(periodics) != 0 {
+		payload["periodics"] = periodics
+	}
+
+	if len(presubmits) == 0 && len(postsubmits) == 0 && len(periodics) == 0 {
+		log.Printf("%v has no forkable configs for version %v\n", repo, version)
+		return
+	}
+
+	err := os.MkdirAll(targetDir, os.ModePerm)
+	if err != nil {
+		log.Fatalf("Couldn't create Directory: %v\n", err)
+	}
+
+	data, err := yaml.Marshal(payload)
+	if err != nil {
+		log.Fatalf("Couldn't marshal presubmits: %v\n", err)
+	}
+
+	newf, err := os.Create(output)
+	if err != nil {
+		log.Fatalf("Couldn't create outputFile: %v\n", err)
+	}
+	defer newf.Close()
+
+	if _, err = newf.Write(data); err != nil {
+		log.Fatalf("Couldn't write to outputFile: %v\n", err)
+	}
+	log.Printf("%v has forked %v Presubmits, %v Postsubmits, %v Periodics for version %v into %v\n",
+		repo,
+		len(presubmits),
+		len(postsubmits),
+		len(periodics),
+		version,
+		output,
+	)
+}
+
+func removeDeprecatedConfigs(repo, baseRepoJobsDirPath string, versions []string) {
+
+	repoString := strings.ReplaceAll(repo, "/", "-")
+	log.Printf("repoString: %v\n", repoString)
+	forkedDir := baseRepoJobsDirPath + "/" + forkDir
+	forkedFiles, err := ghi.GetFileNames(forkedDir, []string{}, false)
+	if err != nil && !errors.Is(err, os.ErrNotExist) {
+		log.Fatalf("Couldn't read from forked files Directory: %v\n", err)
+	}
+	log.Printf("forkedFiles: %v\n", forkedFiles)
+
+	for _, forkedFile := range forkedFiles {
+		if strings.Contains(forkedFile, repoString) {
+			log.Printf("%v matched %v\n", forkedFile, repoString)
+			// branched File belongs to repo
+			matches := false
+			for _, version := range versions {
+				fileVersion := strings.ReplaceAll(version, ".", "-")
+
+				if strings.Contains(forkedFile, repoString+"-"+fileVersion) {
+					// branched File has corresponding branch
+					matches = true
+					break
+				}
+
+			}
+
+			if !matches {
+				// File is deprecated and has no corresponding branch to it anymore
+				log.Printf("Removing %v, because it's config is deprecated\n", forkedFile)
+				os.Remove(forkedFile)
+			}
+		} else {
+			log.Printf("%v didn't match %v\n", forkedFile, repoString)
+		}
+	}
+}
+
+func readReposFromJobs(files []string) []string {
+	repos := []string{}
+	for _, file := range files {
+		j, err := config.ReadJobConfig(file)
+		if err != nil {
+			log.Fatalf("Couldn't read jobConfig: %v\n", err)
+		}
+		for repo := range j.PresubmitsStatic {
+			if contains(repos, repo) {
+				continue
+			} else {
+				repos = append(repos, repo)
+			}
```

Comment:
```suggestion
			if contains(repos, repo) {
				continue
			}
			repos = append(repos, repo)
```

or

```suggestion
			if !contains(repos, repo) {
				repos = append(repos, repo)
			}
```

would be better since continue already terminates current loop